src/Package/Impl/Options/R/RToolsOptionsPage.cs

```diff
@@ -291,18 +294,37 @@ public class RToolsOptionsPage : DialogPage {
             // Save in-memory settings to storage. In case dialog
             // is canceled with some settings modified we will be
             // able to restore them from storage.
-            SaveSettingsToStorage();
+            PersistSettings();
             base.OnActivate(e);
         }
 
         protected override void OnApply(PageApplyEventArgs e) {
             if (e.ApplyBehavior == ApplyKind.Apply) {
                 _applied = true;
                 // On OK persist settings
-                SaveSettingsToStorage();
+                PersistSettings();
                 RtvsTelemetry.Current.ReportSettings();
             }
             base.OnApply(e);
         }
+
+        private void PersistSettings() {
+            SaveSettingsToStorage();
+            _storage.Persist();
+        }
+
+        protected override void LoadSettingFromStorage(PropertyDescriptor prop) {
+            if (_storage.SettingExists(prop.Name)) {
```

Comment:
Yes. I don't want to mess with this code too in 0.5 - it already handles property names. What really should be done is to remove save/load calls from here, have own table of property names for each setting (or use approach similar to `BindableBase.SetProperty` with caller name attribute) and only go via setting storage.  So `RPackage.LoadSettings` won't be calling `RToolsOptionsPage.LoadSettings`